lib/spack/spack/ci/__init__.py

```diff
@@ -69,17 +71,34 @@
 urlopen = web_util.urlopen  # alias for mocking in tests
 
 
-def get_change_revisions():
+@functools.lru_cache(maxsize=32)
+def get_git_root(path) -> Optional[str]:
+    git_exe = spack.util.git.git(required=True)
+    try:
+        print(path)
+        with fs.working_dir(path):
+            # Raises SpackError on command failure
+            git_dir = git_exe(
+                "rev-parse", "--show-toplevel", fail_on_error=True, output=str
```

Comment:
This feature is too new. Use `git rev-parse --git-dir` instead (notice not an abs path and points to the `.git` dir not its parent).

Reply:
It has been around since at least git 2.0.5, which was released in 2014. I can use the other, but why when there is a better option?